File: data/readable_format/cpp/file339.cpp
#include <iostream>

class M1 {
public:
    void bar() {
        std::cout << "In bar" << std::endl;
    }

    void bar_none(int i) {
        std::cout << "In bar_none with i = " << i << std::endl;
    }

    void none_bar(float x) {
        std::cout << "In none_bar with x = " << x << std::endl;
    }
};

int main() {
    M1 m1;
    m1.bar();
    m1.bar_none(5);
    m1.none_bar(3.14f);

    // Optional: Add return 0 at the end of main to indicate successful execution.
    return 0;
}

File: data/readable_format/fortran/file339.f90
module m1
   implicit none
contains
   subroutine bar
      print *, "In bar"
   end subroutine

   subroutine bar_none(i)
      integer :: i
      print *, "In bar_none with i =", i
   end subroutine

   subroutine none_bar(x)
      real :: x
      print *, "In none_bar with x =", x
   end subroutine
end module

program test_m1
    use m1
    implicit none

    call bar
    call bar_none(5)
    call none_bar(3.14)
end program test_m1